store/localcachelayer/user_layer.go

```diff
@@ -39,6 +49,48 @@ func (s LocalCacheUserStore) InvalidatProfileCacheForUser(userId string) {
 	}
 }
 
+func (s LocalCacheUserStore) InvalidateProfilesInChannelCacheByUser(userId string) {
+	keys := s.rootStore.profilesInChannelCache.Keys()
+
+	for _, key := range keys {
+		if cacheItem, ok := s.rootStore.profilesInChannelCache.Get(key); ok {
+			userMap := cacheItem.(map[string]*model.User)
+			if _, userInCache := userMap[userId]; userInCache {
+				s.rootStore.doInvalidateCacheCluster(s.rootStore.profilesInChannelCache, key.(string))
+				if s.rootStore.metrics != nil {
```

Comment:
I think this metric is already added by the `doInvalidateCacheCluster` method.

Reply:
Metrics are not currently being tracked on layer.go. Shall I add the metrics or should I keep it like this?

Reply:
sorry, my bad, you are totally right here. Leave it as is now.